tests/testthat/test-sort_linter.R

```diff
@@ -108,15 +108,19 @@ test_that("sort_linter blocks simple disallowed usages", {
   sorted_msg <- rex::rex("Use !is.unsorted(x) to test the sortedness of a vector.")
 
   expect_lint("sort(x) == x", sorted_msg, linter)
+  expect_lint("identical(x, sort(x))", sorted_msg, linter)
 
   # argument order doesn't matter
   expect_lint("x == sort(x)", sorted_msg, linter)
+  expect_lint("identical(sort(x), x)", sorted_msg, linter)
 
   # inverted version
   expect_lint("sort(x) != x", unsorted_msg, linter)
+  expect_lint("!identical(x, sort(x))", unsorted_msg, linter)
 
   # expression matching
   expect_lint("sort(foo(x)) == foo(x)", sorted_msg, linter)
+  expect_lint("identical(foo(x), sort(foo(x)))", sorted_msg, linter)
```

Comment:
Let's add some more tests against false positives, like `identical(sort(x), y)`, `identical(sort(foo(x)), x)`, and a test that this catches expressions: `identical(sort(foo(x)), foo(x))`